### DISCUSSION TITLE
How to use YARA rules in my C/C++ project?

### DISCUSSION THREAD

**Discussion by MuhabEssam** [VirusTotal/yara#2063] [Q&A]
I download the libyara folder that include c and h. I try to include yara.h in my project but this error found
**yara.h not found**
I solve this by copying the header files to my include folder in Mingw and when i try to build again it gives
**error yr_intilialzie() undefined reference**

**Comment by micromatrix14** [comment]
Undefined reference means you are trying to use a symbol whose definition is missing during the linking stage.
In your case, the translation unit can see the declaration, but the linker cannot find the actual definition.

This means the declarations are visible in other words, you successfully included the necessary header (.h) files. However, you did not include them correctly. You should not copy header files into your compiler’s global include directory. Instead, you should configure your compiler or CMake project to point to the YARA include directory.

Anyway as for the error message, it means you still have not told the linker where to find the definitions for those symbols. You need to link the YARA static library to your application by specifying the library path and the library itself during the linking stage (.lib) files.
